src/superface-test.utils.ts

```diff
@@ -350,3 +357,89 @@ export function checkSensitiveInformation(
     }
   }
 }
+
+function hasProperty<K extends PropertyKey>(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  obj: any,
+  propKey: K
+): obj is Record<K, unknown> {
+  return !!obj && propKey in obj;
+}
+
+function isfunction<R extends unknown>(
+  value: unknown,
+  returnType?: R
+): value is () => R {
+  if (returnType) {
+    return typeof value === 'function' && typeof value() === typeof returnType;
+  }
+
+  return typeof value === 'function';
+}
+
+/**
+ * Checks for structural typing of specified `testInstance` and returns
+ * corresponding instance of hash generator
+ *
+ * It checks for jest's `expect` instance and mocha's `this` instance,
+ * otherwise it generates hash according to specified `testName` or `input` in test run
+ */
+export function getGenerator(testInstance: unknown): IGenerator {
```

Comment:
I miss unit tests of `getGenerator` factory.

Reply:
I'll add them asap. 👍🏻